flexget/plugins/clients/tarantool_queue.py

```diff
@@ -0,0 +1,119 @@
+from loguru import logger
+from flexget.entry import Entry
+from flexget.event import event
+from flexget import plugin
+
+logger = logger.bind(name='tarantool_queue')
+
+class TarantoolTube:
+    def __init__(self, queue, tube_name):
+        self._queue = queue
+        self._tube_name = tube_name
+
+    def put(self, data):
+        return self._queue.call('queue.tube.{0}:put'.format(self._tube_name), data)
+
+    def take(self, timeout=None):
+        entries = self._queue.call('queue.tube.{0}:take'.format(self._tube_name), timeout)
+        if entries is None:
+            entries = []
+        return entries
+
+    def ack(self, task_id):
+        return self._queue.call('queue.tube.{0}:ack'.format(self._tube_name), task_id)
+
+class TarantoolQueue:
+    def __init__(self, tarantool_module, host, port, user, password):
+        self._server = tarantool_module.connect(host, port, user=user, password=password)
+        self._tubes = {}
+
+    def get_tube(self, tube_name):
+        if tube_name in self._tubes:
+            tube = self._tubes[tube_name]
+        else:
+            tube = TarantoolTube(self, tube_name)
+            self._tubes[tube_name] = tube
+        return tube
+
+    def call(self, function, *args):
+        return self._server.call(function, *args)
+
+class PluginTarantoolQueueBase:
+    schema = {
+        'anyOf': [
+            {'type': 'boolean'},
+            {
+                'type': 'object',
+                'properties': {
+                    'host': {'type': 'string'},
+                    'port': {'type': 'integer'},
+                    'user': {'type': 'string'},
+                    'password': {'type': 'string'},
+                    'tube': {'type': 'string'},
+                    'chunk_size': {'type': 'integer', 'default': 30}
+                },
+                'additionalProperties': False,
+            },
+        ]
+    }
+
+    def get_queue(self, tarantool_module, config):
+        queue = TarantoolQueue(tarantool_module,
+                                     config.get('host'),
+                                     config.get('port'),
+                                     config.get('user'),
+                                     config.get('password'))
+
+        return queue
+
+    def get_tube(self, tarantool_module, config):
+        return self.get_queue(tarantool_module, config).get_tube(config.get('tube'))
```

Comment:
I see you're re-using the method name `get_tube` from ` class TarantoolQueue: `. The way this is written makes is look almost recursive. I wonder if it would be better to choose a different name, if you can still have it makes sense,